### DISCUSSION TITLE
CCL dependency on TCA conflicts with dependency on TCA

### DISCUSSION THREAD

**Discussion by ferologics** [pointfreeco/composable-core-location#26] [Q&A]
**Describe the bug**
In projects that depend on both TCA and CCL Xcode raises the following warning during SPM package resolution:
```
'composable-core-location' dependency on 'https://github.com/pointfreeco/swift-composable-architecture' conflicts with dependency on 'https://github.com/pointfreeco/swift-composable-architecture' which has the same identity 'swift-composable-architecture'. this will be escalated to an error in future versions of SwiftPM.
```

**To Reproduce**
1. Create a new project
2. Add TCA and CCL as dependencies
3. Reset package caches
4. Observe warning

**Expected behavior**
No warning during SPM package resolution.

**Environment**
 - Xcode Version `13.3 (13E113)`
 - Swift version `5.6`
 - macOS 12.3 Beta (21E5196i)

**Comment by stephencelis** [comment]
@ferologics How are you depending on TCA? Can you provide the exact git URL and version you're pinning to, etc.?

**Comment by ferologics** [comment]
@stephencelis the url is https://github.com/globekeeper/swift-composable-architecture and we use the [dynamic](https://github.com/globekeeper/swift-composable-architecture/tree/dynamic) branch, which is a fork from version `0.20.0` release with dynamic framework product type (workaround for missing symbols for `OrderedCollections` issue).

**Reply by stephencelis** [reply]
> (workaround for missing symbols for `OrderedCollections` issue).

Can you provide a link to and/or repro for this issue? We've seen certain versions of Xcode have issues with SPM dependencies, but aren't aware of any issue currently affecting TCA in the latest version.

**Reply by ferologics** [reply]
https://github.com/pointfreeco/swift-composable-architecture/discussions/798

**Comment by stephencelis** [comment]
@ferologics Ah, if you're depending on a fork, then you also need to fork any dependency to make it depend on that fork. I'm not aware of SPM supporting the ability to work around this by for example "re-pinning" a dependency's dependency.

Since this isn't a bug in the library, and more a question of how SPM works, I'm going to convert this to a discussion.

**Reply by ferologics** [reply]
@stephencelis the issue is that CCL depends on TCA. Project which includes both TCA & CCL will have this warning from SPM. A more visual way to represent this would be:
```
Project
  | - TCA <------|
  | - CCL        | -- SPM raises a warning about duplicate dependency
    | - TCA <----|
```

The way the warning is phrased suggests that the `dynamic` change on the fork is not responsible for this. It should be easy to try and reproduce in a fresh project to get a confirmation for this.

**Reply by ferologics** [reply]
In other words, fork or not, the conflicting package identity is the issue SPM warns about.
